Adds accessibility notes to breadcrumbs

andresgalante opened the pull request:
This PR adds accessibility remarks on breadcrumbs following WAI-ARIA design pattern:

https://www.w3.org/TR/wai-aria-practices/examples/breadcrumb/index.html

What do you think?

andresgalante committed:
adds accecibility notes to breadcrumbs

mdo commented on docs/4.0/components/breadcrumb.md:
```diff
@@ -32,3 +32,22 @@ Similar to our navigation components, breadcrumbs work fine with or without the
 {% endexample %}
+
+
+## Regarding accessibility
```
This would be the only heading on the page; can we change it to `## Accessibility` for brevity? And then also add an `## Overview` or `## Example` above line 8?

mdo commented on docs/4.0/components/breadcrumb.md:
```diff
@@ -32,3 +32,22 @@ Similar to our navigation components, breadcrumbs work fine with or without the
+
+## Regarding accessibility
+
+If you’re using breadcrumbs to provide a navigation, be sure to add a `aria-label="breadcrumb"` to describe the type of navigation provided in the `<nav>` element.
```
Should we be doing this in our examples?

andresgalante replied:
I think that if we can deliver accessible snippets from bootstrap we will be doing a huge dent to help people with disabilities.

So even though the other examples will render correctly, I think we should deliver just one example that follows an accessible pattern.

I'll change this PR and send a new option

mdo commented on docs/4.0/components/breadcrumb.md:
```diff
@@ -32,3 +32,22 @@ Similar to our navigation components, breadcrumbs work fine with or without the
+
+Also apply an `aria-current="page"` to the last `<a>` of the set to indicate that it represents the current page.
+
+More information on [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices/#breadcrumb).
```
Feel like the above three lines should be a single paragraph vs three separate lines.

andresgalante commented:
Hi @mdo 
I've reduce the accessibility text, removed the non accessible code snippets and replaced them for the recommended W3C design pattern.

As I wrote on the comment above, I think that the more accessibility we can deliver built into bootstrap components the better the web will be.

What do you think?

patrickhlauke commented on docs/4.0/components/breadcrumb.md:
```diff
@@ -4,31 +4,36 @@ title: Breadcrumb
+
+## Accessibility
+
+Since breadcrumbs provide a navigation, be sure to add a `aria-label="breadcrumb"` to describe the type of navigation provided in the `<nav>` element. Also apply an `aria-current="page"` to the last item of the set to indicate that it represents the current page.
```
I'd soften the language from "be sure to add" to something more like "it's a good idea to add". As it's up to the author to assign a label (i.e. "breadcrumb" isn't a mandated value, just a suggestion), I'd further change it to something like "it's a good idea to add a meaningful label such as `aria-label="Breadcrumb"`.

patrickhlauke reviewed (changes requested):
Barring the slight rewording I suggested, this looks good to me 👍

andresgalante commented:
Hey @patrickhlauke, that makes a lot of sense, thanks for the review.

I've also changed the second phrase. Can you please take another look at it?

patrickhlauke commented on docs/4.0/components/breadcrumb.md:
```diff
@@ -4,31 +4,38 @@ title: Breadcrumb
+
+
+
+More information on [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices/#breadcrumb).
```
sorry, one final change request (missed it first time around): possibly reword to "For more information, see the [WAI-ARIA Authoring Practices for the breadcrumb pattern](...)." (and maybe remove two of the preceding line breaks, to keep it nice and tight)

andresgalante replied:
@patrickhlauke Nothing to be sorry about, I'll update it now. I am not an english native speaker, so any help on wording is always welcome. I'll update the PR now

andresgalante commented:
Hi @patrickhlauke the change is done, let me know if I am missing anything else,
thanks a lot for taking the time to review this one.

patrickhlauke commented:
All good for me, thanks :)
